File: golem-hello-world-py/requestor.py
#!/usr/bin/env python3
import asyncio
from typing import AsyncIterable

from yapapi import Golem, Task, WorkContext
from yapapi.log import enable_default_logger
from yapapi.payload import vm


async def worker(context: WorkContext, tasks: AsyncIterable[Task]):
    async for task in tasks:
        script = context.new_script()

        # upload & run the provider.py script
        script.upload_file("provider.py", "/golem/input/provider.py")
        future_result = script.run("/bin/sh", "-c", "python3 /golem/input/provider.py")

        yield script

        task.accept_result(result=await future_result)


async def main():
    package = await vm.repo( 
        image_hash="eb26fc55959db9901148fc6dfaed8578b2f8fb7067c88feaeb1846f8",
    )

    tasks = [Task(data=None)]

    async with Golem(budget=1.0, subnet_tag="devnet-beta") as golem:
        async for completed in golem.execute_tasks(worker, tasks, payload=package):
            # print out the console output
            print(completed.result.stdout)


if __name__ == "__main__":
    enable_default_logger(log_file="out.log")

    loop = asyncio.get_event_loop()
    task = loop.create_task(main())
    loop.run_until_complete(task)

File: golem-hello-world-js/requestor.js
const { Executor, Task, utils: { asyncWith }, vm } = require("yajsapi");

async function main() {
  const package = await vm.repo({
    image_hash: "48d399e06dba275294b642c4d1c9de10d84186ef0e6f91371b9d5fd2"
  });
  const tasks = [new Task({})];

  async function* worker(context, tasks) {
    for await (let task of tasks) {
      // upload & run the provider.js file

      context.send_file("provider.js", "/golem/input/provider.js");
      context.run("/bin/sh", ["-c", "node /golem/input/provider.js"]);

      const future_result = yield context.commit();
      const { results } = await future_result;
      task.accept_result(results[results.length - 1])
    }
  }

  await asyncWith(
    new Executor({ task_package: package, budget: "1.0", subnet_tag: "devnet-beta.2" }),
    async (executor) => {
      for await (let completed of executor.submit(worker, tasks)) {
        // console.log the console output of the provider machine
        console.log(completed.result().stdout);
      }
    }
  );
}

main();